dynamiq/connections/connections.py

```diff
@@ -929,3 +930,32 @@ class DeepSeek(BaseApiKeyConnection):
 
     def connect(self):
         pass
+
+
+class Exa(HttpApiKey):
+    """
+    Represents a connection to the Exa AI Search API.
+
+    Attributes:
+        type (Literal[ConnectionType.Exa]): The type of connection, which is always 'Exa'.
+        url (str): The URL of the Exa API.
+        api_key (str): The API key for authentication, fetched from the environment variable 'EXA_API_KEY'.
+        method (Literal[HTTPMethod.POST]): HTTP method used for the request, defaults to HTTPMethod.POST.
+        headers (dict[str, Any]): HTTP headers for the request.
+    """
+
+    type: Literal[ConnectionType.Exa] = ConnectionType.Exa
+    url: str = Field(default="https://api.exa.ai/search")
+    api_key: str = Field(default_factory=partial(get_env_var, "EXA_API_KEY"))
+    method: Literal[HTTPMethod.POST] = HTTPMethod.POST
+    headers: dict[str, Any] = Field(default_factory=dict)
```

Comment:
Why do we need `url`, `method` and `headers` here?

Reply:
`URL`, `method`, and `headers` are always the same.

Reply:
Structure is the same as for our other tools like `Tavily`, `Zenrows` , etc

Reply:
There is no point in following the same structure and bringing some unused fields to connection.

Reply:
I haven't checked the `Tavily`, `Zenrows` API docs, but I think we might do that incorrectly because, usually, it's how APIs are defined.

Reply:
I've removed `headers` but kept `url`/`method`, you can use either `url` for `scrape` and so on

Reply:
We cannot use different URLs. The node is working with the exact URL and exact method.

Reply:
Please inherit from `Http` instead of `HttpApiKey`

Reply:
if url unchanged please use Literal instead of str